terraform/monitoring/01_uptime_checks.tf

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Here we create a default HTTP Uptime Check on the External IP of our
+# Kubernetes cluster revealed by the Load Balancer
+# Our uptime check has a period of 1 minute and times out after 10 seconds
+resource "google_monitoring_uptime_check_config" "http" {
+  display_name = "HTTP Uptime Check"
+  timeout      = "10s"
+  period       = "60s"  
+
+  http_check {
+    path = "/"
+    port = "80"
+  }
+
+  monitored_resource {
+    type = "uptime_url"
+    labels = {
+      project_id = "${var.project_id}"
+      host       = "${var.external_ip}"
+    }
+  }
+}
+
+# Here we place an alerting policy on the HTTP Uptime Check that alerts based on
+# failed uptime checks. The alerting policy will be triggered if the uptime check
+# fails more than 1 time over a 1 minute period aligned over 20 minutes.
```

Comment:
Right sorry my wording here isn't clear - I'll adjust it. To answer your question: the uptime check will check once every minute with a timeout of 10 seconds. If there is more than one failure (more than one region fails) it will send an alert. The 20 minute alignment is for the metric that is displayed, it shows the data points over the past 20 minutes. So our uptime check is rather strict.